_posts/2020-05-30-Metal3-Cluster-Deployment.md

```diff
@@ -0,0 +1,544 @@
+- The script calls an ansible playbook with necessary parameter ( from env vars and defaults )
+- The playbook executes the role -, [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), which runs the corresponding [task_file](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/tasks) for provisioning/deprovisioning the cluster/controlplane or a worker
+- There are [templates](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test/templates) in the role, which are copied to the `manifest` directory. They are used for generating configuration for the cluster and kubeadm, which is then supplied to the Kubernetes module of ansible to create the cluster.
+- During provisioning, first the `clusterctl` env file is generated, then and `manifest` file is created/updated using the `clusterctl` tool, which is then supplied to `kustomize` tool to create the config files for provisioning the resource, from the available templates.
+- The manifests are generated for - cluster, controlplane and machinedeployment, and the worker manifests are merged with the deployment manifests.
+- Finally a yaml file is generated with all the details for the cluster/controlplane/worker, which is then passed to K8s module in ansible which creates the resources.
+- There are also manifests stored in ${Manifests} directory - [`${metal3-dev-env}/vm-setup/roles/v1aX_integration_test/files/manifests`](https://github.com/metal3-io/metal3-dev-env/tree/master/vm-setup/roles/v1aX_integration_test), for the cluster which can be used to deprovision the cluster at a later point in time.
+> **Note** : * the `manifests` directory is not present but created when you first use the scripts.*
+- Centos or Ubuntu images [are downloaded](https://github.com/metal3-io/metal3-dev-env/blob/master/vm-setup/roles/v1aX_integration_test/tasks/download_image.yml) when provisioning a controlplane or a worker
+
+**Note** : *Here is a depiction of the common steps, mainly involving generating templates and config files.*
+
+![A diagram depicting the Generate Templates workflow](/assets/images/metal3-generate-templates.svg)
+
+
+### Provision Cluster
+This script, located at the path - `${metal3-dev-env}/scripts/v1alphaX/provision_clusters.sh`, provisions the cluster. This script creates a `Metal3Cluster` resource. To see if you have a successful deployment, just do - 
+```console
+kubectl get metal3cluster ${CLUSTER_NAME} -n metal3
+```
+This will return the cluster deployed, and you can check the cluster details by describing the returned resource.
+
+Here is what a `Cluster` resource looks like :
+```console
+kubectl describe Cluster ${CLUSTER_NAME} -n metal3
+```
+```yaml
+Name:         eko
+Namespace:    metal3
+Labels:       <none>
+Annotations:  <none>
+API Version:  cluster.x-k8s.io/v1alpha3
+Kind:         Cluster
+Metadata:
```

Comment:
That's a long content and several things don't bring much... maybe shorten it a bit by replacing some parts, like the whole `Metadata` content with something like `[...snip...]`

Then maybe add a couple of lines explaining the highlights. Probably worth pointing at the control plane endpoint and explain that this is where your new cluster's API will be available